src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -706,6 +708,13 @@ private function createHashers(array $hashers, ContainerBuilder $container): voi
         $hasherMap = [];
         foreach ($hashers as $class => $hasher) {
             $hasherMap[$class] = $this->createHasher($hasher);
+            $id = 'security.password_hasher.' .ContainerBuilder::hash($class);
+            $container
+                ->register($id, PasswordHasherInterface::class)
+                ->setFactory([new Reference('security.password_hasher_factory'), 'getPasswordHasher'])
+                ->setArgument(0, $class)
+            ;
+            $container->registerAliasForArgument($id, PasswordHasherInterface::class, $class);
```

Comment:
shouldn't we skip doing this when $class is effectively a FQ[CI]N?

Reply:
good idea